Enable writing column names with mixed dtype in parquet writer when `mode.pandas_compatible=True` (#13505)
rapidsai/cudf: docs/cudf/source/user_guide/pandas-comparison.md

Unlike Pandas, cuDF does not support duplicate column names.
It is best to use unique strings for column names.

## Writing dataframe to parquet with mixed column names

When there is a dataframe with mixed column names, pandas type-casts each

> column name to `string` before writing to parquet file. `cudf` raises an

Reviewer: ```suggestion
column name to `str` before writing to a Parquet file. `cudf` raises an
```

error by default if this is attempted. However, to achieve similar behavior

> as pandas you can enable pandas compatibility mode option, which will

Reviewer: ```suggestion
as pandas you can enable the `mode.pandas_compatible` option, which will
```

Reviewer: Please check that code snippets are using the right number of backticks. I think one backtick just italicizes the text…

Author: Verified that one backtick is bringing the code font from the deployed docs: https://docs.rapids.ai/api/cudf/nightly/user_guide/pandas-comparison/#no-true-object-data-type

Reviewer: Of course. This is Markdown and not RST. My fault.

enable `cudf` to cast the column names to `str` just like pandas.

```python
>>> import cudf

> >>> df = cudf.DataFrame({1: [1, 2, 3], "1": ["a", "b", "c"]})

Author: ```suggestion
>>> df = cudf.DataFrame({1: [1, 2, 3], 2: ["a", "b", "c"]})
```

>>> df.to_parquet("df.parquet")

Traceback (most recent call last):

> ValueError: parquet must have string column names

Author: Done 👍

> ValueError: parquet must have string column names

Author: ```suggestion
ValueError: Writing a Parquet file requires string column names
```

>>> cudf.set_option("mode.pandas_compatible", True)
>>> df.to_parquet("df.parquet")

> UserWarning: The DataFrame has column names of mixed type. They will be converted to strings and not roundtrip correctly.

Reviewer: “Round trip” is a little bit of an informal term, I think? It may not be needed here.
```suggestion
UserWarning: The DataFrame has column names of non-string type. They will be converted to strings on write.
```
